Usiqee/Module/Auth/PreAuthVC.swift

```diff
@@ -76,6 +112,59 @@ extension PreAuthVC {
     @IBAction func onDismissToggle() {
         dismiss(animated: true, completion: nil)
     }
+    
+    @available(iOS 13, *)
+    @IBAction func signInWithAppleToggle() {
+        let nonce = String.randomNonceString()
+        currentNonce = nonce
+        let appleIDProvider = ASAuthorizationAppleIDProvider()
+        let request = appleIDProvider.createRequest()
+        request.requestedScopes = [.fullName, .email]
+        request.nonce = sha256(nonce)
+        
+        let authorizationController = ASAuthorizationController(authorizationRequests: [request])
+        authorizationController.delegate = self
+        authorizationController.presentationContextProvider = self
+        authorizationController.performRequests()
+        signInWithAppleButton.loadingIndicator(show: true)
+    }
+    
+    @available(iOS 13, *)
+    private func sha256(_ input: String) -> String {
+        let inputData = Data(input.utf8)
+        let hashedData = SHA256.hash(data: inputData)
+        let hashString = hashedData.compactMap {
+            return String(format: "%02x", $0)
+        }.joined()
+        
+        return hashString
+    }
```

Comment:
this can be a string extension, what de you think? `nonce.sha256`

Reply:
Done